tango/common/lazy.py

```diff
@@ -85,3 +86,54 @@ def construct(self, **kwargs) -> T:
         # this will overwrite the ones in self._constructor_extras with what's in kwargs.
         constructor_kwargs = {**self._constructor_extras, **kwargs}
         return self.constructor(**constructor_kwargs)
+
+    def det_hash_object(self) -> Any:
+        from tango.common.from_params import FromParams
+
+        class_to_construct: Union[Type[T], Callable[..., T]] = self._constructor
+        if isinstance(class_to_construct, type) and issubclass(class_to_construct, FromParams):
+            params = copy.deepcopy(self._params)
+            if params is None:
+                params = Params({})
+            elif isinstance(params, str):
+                params = Params({"type": params})
+            elif isinstance(params, dict):
+                params = Params(params)
+
+            if not isinstance(params, Params):
+                return None
```

Comment:
Gotcha. How about:
```suggestion
                params = Params(params)
            elif not isinstance(params, Params):
                return None
```

I think that's a little more clear. Just a suggestion though.